src/CMakeLists.txt

```diff
@@ -141,3 +141,21 @@ target_link_libraries(${project_name} ${global_libraries})
 install(TARGETS ${project_name}
   RUNTIME DESTINATION bin
 )
+
+find_package(Plantuml)
+if(PLANTUML_FOUND)
+#message("plantuml found PLANTUML_JARFILE=${PLANTUML_JARFILE}")
+else(Plantuml_FOUND)
+message("plantuml not found. No UML diagrams are generated. You may define the environment variable PLANTUML_PATH...")
```

Comment:
with find_package(PlantUml) plantuml is detected and later inserted into the Doxyfile (generated form Doxyfile.in). 
As noted above, if it is not detected UML diagrams will just be omitted.